client/components/async-load/index.jsx

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { omit } from 'lodash';
+
+export default class AsyncLoad extends Component {
+	static propTypes = {
+		require: PropTypes.func.isRequired,
+		placeholder: PropTypes.node
+	};
+
+	constructor() {
+		super( ...arguments );
+
+		this.state = {
+			component: null
+		};
+	}
+
+	componentWillMount() {
```

Comment:
It also raises the point that the Babel plugin transforms the string into a function which created new for every render. I might experiment a bit to see how difficult it would be to pull the function to the top of the transformed module so that we can do a strict equality check in `componentWillReceiveProps` before calling the `require` prop.